### PR TITLE
OSDOCS#10502: Cluster extension status message improvements

### PR DESCRIPTION
**Pull request by michaelryanpeter**
Version(s): 4.16

Issue: [OSDOCS-10502](https://issues.redhat.com//browse/OSDOCS-10502) and https://issues.redhat.com/browse/OSDOCS-9021

Link to docs preview: 
* https://77274--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/olm_v1/olmv1-installing-an-operator-from-a-catalog.html#olmv1-installing-an-operator_olmv1-installing-operator
* https://77274--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/olm_v1/olmv1-installing-an-operator-from-a-catalog.html#olmv1-updating-an-operator_olmv1-installing-operator

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/olmv1-installing-an-operator.adoc [hunk 1]
```diff
@@ -87,14 +88,29 @@ select( .schema == "olm.channel" ) | select( .name == "latest" ) | \
...
+"openshift-pipelines-operator-rh.v1.12.2"
+"openshift-pipelines-operator-rh.v1.13.0"
+"openshift-pipelines-operator-rh.v1.14.1"
+"openshift-pipelines-operator-rh.v1.14.2"
+"openshift-pipelines-operator-rh.v1.14.3"
+"openshift-pipelines-operator-rh.v1.14.4"
+
 ----
 ====
+. If you want to install your extension into a new namespace, run the following command.
```
Comment (adellape) [inline modules/olmv1-installing-an-operator.adoc:108]:
```suggestion
. If you want to install your extension into a new namespace, run the following command:
```

**Review by jianzhangbjz — CHANGES_REQUESTED** [review]

**Comment by michaelryanpeter** [comment]
Thank you, @jianzhangbjz. Comments addressed in https://github.com/openshift/openshift-docs/pull/77274/commits/7852f3f54dba3399da5969fb77eec4548ea7b485

File: modules/olmv1-installing-an-operator.adoc [hunk 2]
```diff
@@ -142,48 +160,90 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
...
-  installedBundleResource: registry.redhat.io/openshift-pipelines/pipelines-operator-bundle@sha256:e09d37bb1e754db42324fd18c1cb3e7ce77e7b7fcbf4932d0535391579938280
-  resolvedBundleResource: registry.redhat.io/openshift-pipelines/pipelines-operator-bundle@sha256:e09d37bb1e754db42324fd18c1cb3e7ce77e7b7fcbf4932d0535391579938280
+  resourceVersion: ""
 ----
+where:
+
+`spec.channel`:: Displays the channel defined in the CR of the extension.
+`spec.version`:: Displays the version or version range defined in the CR of the extension.
+`status.conditions`:: Displays information about the status and health of the extension.
+`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
```
Comment (varshaprasad96) [inline modules/olmv1-installing-an-operator.adoc:239]:
```suggestion
`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that the bundle which is to be installed is not deprecated.
```

I'm not sure if "extension" is deprecated would be right. It's more on the lines of - while extension tries to install a bundle, the bundle is itself deprecated. I'll leave the wording to you, as you feel it fit :)

Comment (joelanford) [reply modules/olmv1-installing-an-operator.adoc:239]: For `type: Deprecated`, could we somehow say it's a rollup of `PackageDeprecated`, `ChannelDeprecated`, and `BundleDeprecated`, and let the other descriptions do the heavy lifting?

File: modules/olmv1-installing-an-operator.adoc [hunk 3]
```diff
@@ -142,48 +160,90 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
...
+`type: PackageDeprecated`:: Displays if the package is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
+`type: ChannelDeprecated`:: Displays if the channel is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
+`type: BundleDeprecated`:: Displays if the channel is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
```
Comment (joelanford) [inline modules/olmv1-installing-an-operator.adoc:242]: Copy pasta

```suggestion
`type: BundleDeprecated`:: Displays if the bundle is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
```

File: modules/olmv1-installing-an-operator.adoc (hunk 2 above)
Comment (michaelryanpeter) [reply modules/olmv1-installing-an-operator.adoc:239]: @varshaprasad96 would something like this make sense?
```suggestion
`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that the resolved bundle is not deprecated.
```

**Comment by michaelryanpeter** [comment]
@joelanford and @varshaprasad96 Thanks for the feedback! I addressed your comments in https://github.com/openshift/openshift-docs/pull/77274/commits/d68653e19ebbf0eb07ade4d39acff7f91d0b04c4

I think that this will be clearer in 4.17 where I plan on making a `status.conditions` table. Let me know what you think.

File: modules/olmv1-installing-an-operator.adoc [hunk 4]
```diff
@@ -142,48 +160,90 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
...
-  installedBundleResource: registry.redhat.io/openshift-pipelines/pipelines-operator-bundle@sha256:e09d37bb1e754db42324fd18c1cb3e7ce77e7b7fcbf4932d0535391579938280
-  resolvedBundleResource: registry.redhat.io/openshift-pipelines/pipelines-operator-bundle@sha256:e09d37bb1e754db42324fd18c1cb3e7ce77e7b7fcbf4932d0535391579938280
+  resourceVersion: ""
 ----
+where:
+
+`spec.channel`:: Displays the channel defined in the CR of the extension.
+`spec.version`:: Displays the version or version range defined in the CR of the extension.
+`status.conditions`:: Displays information about the status and health of the extension.
+`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that none ofthe `reason: Deprecated` conditions are marked as deprecated.
```
Comment (michaelryanpeter) [inline modules/olmv1-installing-an-operator.adoc:239]:
```suggestion
`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that none of the other `reason: Deprecated` conditions are marked as deprecated.
```

**Comment by michaelryanpeter** [comment]
Thank you for the feedback, @jianzhangbjz , @joelanford and @varshaprasad96 

I have addressed your collective comment` in this commit: https://github.com/openshift/openshift-docs/pull/77274/commits/95125c985f1a2353e8dd69529116d4bc4a42b1e3

File: modules/olmv1-installing-an-operator.adoc [hunk 5]
```diff
@@ -142,48 +160,95 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
...
+      kubectl.kubernetes.io/last-applied-configuration: |
+        {"apiVersion":"olm.operatorframework.io/v1alpha1","kind":"ClusterExtension","metadata":{"annotations":{},"name":"pipelines-operator"},"spec":{"channel":"latest","installNamespace":"openshift-marketplace","packageName":"openshift-pipelines-operator-rh","pollInterval":"30m"}}
+    creationTimestamp: "2024-06-10T17:50:51Z"
+    generation: 1
+    name: pipelines-operator
+    resourceVersion: "53324"
+    uid: c54237be-cde4-46d4-9b31-d0ec6acc19bf
+  spec:
+    channel: latest
+    installNamespace: openshift-marketplace
```
Comment (kuiwang02) [inline modules/olmv1-installing-an-operator.adoc:183]: @michaelryanpeter suggest openshift-operators, not openshift-marketplace.
per my understanding, openshift-marketplace is used for catsrc, while openshift-operators is used for operator installation.

should check it in the PR.

Comment (michaelryanpeter) [reply modules/olmv1-installing-an-operator.adoc]: @kuiwang02 Thank you for the feedback. Addressed in https://github.com/openshift/openshift-docs/commit/587f6bd25e3cad42aefa47f211480b2064cb1fb4.

File: modules/olmv1-installing-an-operator.adoc [hunk 6]
```diff
@@ -142,48 +160,95 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
...
-apiVersion: olm.operatorframework.io/v1alpha1
-kind: ClusterExtension
+apiVersion: v1
+items:
+- apiVersion: olm.operatorframework.io/v1alpha1
+  kind: ClusterExtension
+  metadata:
+    annotations:
+      kubectl.kubernetes.io/last-applied-configuration: |
+        {"apiVersion":"olm.operatorframework.io/v1alpha1","kind":"ClusterExtension","metadata":{"annotations":{},"name":"pipelines-operator"},"spec":{"channel":"latest","installNamespace":"openshift-marketplace","packageName":"openshift-pipelines-operator-rh","pollInterval":"30m"}}
```
Comment (kuiwang02) [inline modules/olmv1-installing-an-operator.adoc:175]: @michaelryanpeter it is not changed here `"installNamespace":"openshift-marketplace"`.

**Review by varshaprasad96 — APPROVED** [review]
The changes look good to me!

/lgtm

**Comment by kuiwang02** [comment]
i am ok

**Pull request merged**